README.md

```diff
@@ -1,2 +1,440 @@
+Robot Framework has easy syntax, utilising human-readable keywords and its capabilities can be extended by libraries implemented with Python or Java.
+
+This BrowserStack Example repository demonstrates a Selenium test framework written in Robot Framework with parallel testing capabilities. The Selenium test scripts are written for the open source [BrowserStack Demo web application](https://bstackdemo.com) ([Github](https://github.com/browserstack/browserstack-demo-app)). This BrowserStack Demo App is an e-commerce web application which showcases multiple real-world user scenarios. The app is bundled with offers data, orders data and products data that contains everything you need to start using the app and run tests out-of-the-box.
+
+The Selenium test tests are run on different platforms like on-prem, docker and BrowserStack using various run configurations and test capabilities.
+
+---
+
+## Repository setup
+
+- Clone the repository
+
+- Ensure you have the following dependencies installed on the machine
+    - Python 3
+
+    Install the requirements:
+    ```sh
+    pip install -r requirements.txt
+    ```
+
+
+## About the tests in this repository
+
+  This repository contains the following Selenium tests:
+
+  | Module   | Test name                          | Description |
+  | ---   | ---                                   | --- |
+  | E2E      | E2E Test                | This test scenario verifies successful product purchase lifecycle end-to-end. It demonstrates the [Page Object Model design pattern](https://www.browserstack.com/guide/page-object-model-in-selenium) and is also the default test executed in all the single test run profiles. |
+  | Login    | Check if Signin opens on clicking on favourites nav item          | This test verifies the login workflow with different types of valid login users. |
+  | Login    | Check Login with locked_user               | This test verifies the login workflow error for a locked user. |
+  | Offers   | Set GPS location to Mumbai and check offers     | This test mocks the GPS location for Mumbai and verifies that the product offers applicable for the Mumbai location are shown.   |
+  | Product  | Apply Apple And Samsung Filter          | This test verifies that the Apple products are only shown if the Apple vendor filter option is applied. |
+  | Product  | Apply 'Lowest to Highest' Order By Filter   | This test verifies that the product prices are in ascending order when the product sort "Lowest to Highest" is applied. |
+  | User     | Check Login with image_not_loading_user | This test verifies that the product images load for user: "image_not_loading_user" on the e-commerce application. Since the images do not load, the test case assertion fails.|
+  | User     | Check Order in existing_orders_user |  This test verifies that existing orders are shown for user: "existing_orders_user"  |
```

Comment:
if you are allowing the test execution based on test suites, please update the module name in the table to the exact value required to run the single test.
if you are allowing the test execution based on exact test names, please update the test name in the table to the exact value required to run the single test.

In the sections below, the same will be referred
for e.g. 

 where,  the argument 'test-name' can be any RobotFramework scenario name configured in this repository.

  E.g. "Check Login with locked_user", "E2E Test", "Apply 'Lowest to Highest' Order By Filter" or any of the other test scenario names, as outlined in [About the tests in this repository](#About-the-tests-in-this-repository) section.

[ Not sure the names are right above, you must verify ]